onboarding/main.go

```diff
@@ -57,90 +56,87 @@ func main() {
 	// Instead, a safer approach is to delete both secrets and then recreate them simultaneously
 	// to ensure consistency and accuracy of all secrets. By this way it will be easier to diagnose the
 	// issues if one or two secrets do not exist instead of trying to understand if they match
-	err = clientset.CoreV1().
-		Secrets(operatorNamespace).
-		Delete(ctx, onboardingValidationPrivateKeySecretName, metav1.DeleteOptions{})
+	privateSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      onboardingValidationPrivateKeySecretName,
+			Namespace: operatorNamespace,
+		},
+	}
+	err = cl.Delete(ctx, privateSecret)
 	if err != nil && !kerrors.IsNotFound(err) {
 		klog.Errorf("failed to delete private secret: %v", err)
 		os.Exit(1)
 	}
 
 	// Delete public key secret
-	err = clientset.CoreV1().
-		Secrets(operatorNamespace).
-		Delete(ctx, onboardingValidationPublicKeySecretName, metav1.DeleteOptions{})
+	publicSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      onboardingValidationPublicKeySecretName,
+			Namespace: operatorNamespace,
+		},
+	}
+	err = cl.Delete(ctx, publicSecret)
 	if err != nil && !kerrors.IsNotFound(err) {
 		klog.Errorf("failed to delete public secret: %v", err)
 		os.Exit(1)
 	}
 
-	storageClusterMetadata, err := getStorageClusterMetadata(ctx, operatorNamespace, clientset)
+	storageClusterMetadata, err := getStorageClusterMetadata(ctx, operatorNamespace, cl)
 	if err != nil {
 		klog.Errorf("failed to get storage cluster metadata: %v", err)
 		os.Exit(1)
 	}
 
-	privateSecret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      onboardingValidationPrivateKeySecretName,
-			Namespace: operatorNamespace,
-			OwnerReferences: []metav1.OwnerReference{
-				{
-					UID:        storageClusterMetadata.UID,
-					APIVersion: storageClusterMetadata.APIVersion,
-					Kind:       storageClusterMetadata.Kind,
-					Name:       storageClusterMetadata.Name,
-				},
-			}},
-		StringData: map[string]string{
-			"key": privatePem,
+	privateSecret.ObjectMeta.OwnerReferences = []metav1.OwnerReference{
+		{
+			UID:        storageClusterMetadata.UID,
+			APIVersion: storageClusterMetadata.APIVersion,
+			Kind:       storageClusterMetadata.Kind,
+			Name:       storageClusterMetadata.Name,
 		},
 	}
```

Comment:
Why don't just use `SetOwnerReference` the same way we use in other places in this code base
(e.g. https://github.com/red-hat-storage/ocs-operator/blob/c8a0b8fb88016821f51605aad025f33552307a8b/controllers/storagecluster/provider_server.go#L140)